packages/host/app/resources/auto-attached-card.ts

```diff
@@ -1,40 +1,86 @@
-import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 
 import { Resource } from 'ember-resources';
 
+import { TrackedSet } from 'tracked-built-ins';
+
+import type { StackItem } from '@cardstack/host/lib/stack-item';
+
 import { type CardDef } from 'https://cardstack.com/base/card-api';
 
 interface Args {
   named: {
-    lastTopMostCard: CardDef | undefined; // top-most card from the card stack
+    stackItems: StackItem[];
     attachedCards: CardDef[] | undefined; // cards manually attached in ai panel
   };
 }
 
 /**
- * Manages the auto-attachment of the top-most card in consideration of user-actions of manually
+ * Manages the auto-attachment of cards within our stack in consideration of user-actions of manually
  * removing and attaching new cards in the ai panel
  */
-
 export class AutoAttachment extends Resource<Args> {
-  @tracked card: CardDef | undefined;
-  private lastCard: CardDef | undefined;
+  cards: TrackedSet<CardDef> = new TrackedSet(); // auto-attached cards
+  private lastStackedItems: StackItem[] = [];
+  private lastRemovedCards: Set<string> = new Set(); // internal state, changed from the outside. It tracks, everytime a card is removed in the ai-panel
 
   modify(_positional: never[], named: Args['named']) {
-    const { lastTopMostCard, attachedCards } = named;
-    if (lastTopMostCard === undefined) {
-      this.card = undefined;
-      this.lastCard = undefined;
-      return;
+    const { stackItems, attachedCards } = named;
+    if (this.stackItemsChanged(stackItems)) {
+      // we must be sure to clear the lastRemovedCards state so cards can be auto-attached again
+      // note: if two of the same cards are opened on separate stack, one will be auto-attached.
+      // If one is removed from one of the stacks, the card WILL be auto-attached.
+      this.lastRemovedCards.clear();
+    }
+    this.cards.clear();
+    stackItems.forEach((item) => {
+      if (item === undefined) {
+        // TODO: another place where stackItems = [undefined]. Pls remove after finding root cause
+        return;
+      }
```

Comment:
bleurgh. This happened in test somewhere. I will check it out separately

Reply:
Fixed [here](https://github.com/cardstack/boxel/pull/1210/commits/a0ab6ca0fe99044f0ea951ec0edb5a854293c0d4)